src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.Constructor.cs

```diff
@@ -2056,7 +2056,7 @@ public class Point_MembersHave_JsonIgnore : ITestClass
         public void Verify()
         {
             Assert.Equal(0, X);
-            Assert.Equal(0, Y); // We don't set parameter default value here.
+            Assert.Equal(5, Y);
```

Comment:
So it seems like some of the changes here are meant to address `defaultValue` handling in the case of `JsonIgnoreAttribute`? Is this the expected behavior when `Y` doesn't have `JsonIgnore` and we deserialize `{ "X" : 1 }`?

It's clearly something we should be fixing but I think it's required to fix the bug reported in #59936. I would recommend splitting this PR into one that applies a minimal fix for #599396 for GA backporting and a separate one that addresses default value handling in 7.

Reply:
> Is this the expected behavior when Y doesn't have JsonIgnore and we deserialize { "X" : 1 }?

Yup. Created an issue to track the correct default value handling for 7.0 and reversed related changes - https://github.com/dotnet/runtime/issues/60082.